framework/tlvf/AutoGenerated/src/CmduMessageRx.cpp

```diff
@@ -34,12 +73,174 @@ CmduMessageRx::~CmduMessageRx()
+        }
+        case (4):{
+        return addClass<tlvDeviceBridgingCapability>();
+        }
+        case (6):{
+        return addClass<tlvNon1905neighborDeviceList>();
+        }
+        case (7):{
+        return addClass<tlv1905NeighborDevice>();
+        }
+        case (8):{
+        return addClass<tlvLinkMetricQuery>();
+        }
+        case (9):{
+        return addClass<tlvTransmitterLinkMetric>();
+        }
+        case (10):{
+        return addClass<tlvReceiverLinkMetric>();
+        }
+        case (11):{
+        return addClass<tlvVendorSpecific>();
+        }
+        case (12):{
+        return addClass<tlvLinkMetricResultCode>();
+        }
+        case (13):{
+        return addClass<tlvSearchedRole>();
+        }
+        case (14):{
+        return addClass<tlvAutoconfigFreqBand>();
+        }
+        case (15):{
+        return addClass<tlvSupportedRole>();
+        }
+        case (16):{
+        return addClass<tlvSupportedFreqBand>();
+        }
+        #if 1
+        case (17): {
+            return parseWscTlv();
+        }
+        #endif
+        case (18):{
+        return addClass<tlvPushButtonEventNotification>();
+        }
+        case (19):{
+        return addClass<tlvPushButtonJoinNotification>();
+        }
+        case (128):{
+        return addClass<wfa_map::tlvSupportedService>();
+        }
+        case (129):{
+        return addClass<wfa_map::tlvSearchedService>();
+        }
+        case (130):{
+        return addClass<wfa_map::tlvApRadioIdentifier>();
+        }
+        case (133):{
+        return addClass<wfa_map::tlvApRadioBasicCapabilities>();
+        }
+        case (139):{
+        return addClass<wfa_map::tlvChannelPreference>();
+        }
+        case (140):{
+        return addClass<wfa_map::tlvRadioOperationRestriction>();
+        }
+        case (141):{
+        return addClass<wfa_map::tlvTransmitPowerLimit>();
+        }
+        case (142):{
+        return addClass<wfa_map::tlvChannelSelectionResponse>();
+        }
+        case (143):{
+        return addClass<wfa_map::tlvOperatingChannelReport>();
+        }
+        case (146):{
+        return addClass<wfa_map::tlvClientAssociationEvent>();
+        }
+        case (147):{
+        return addClass<wfa_map::tlvApMetricQuery>();
+        }
+        case (155):{
+        return addClass<wfa_map::tlvSteeringRequest>();
+        }
+        case (156):{
+        return addClass<wfa_map::tlvSteeringBTMReport>();
+        }
+        case (157):{
+        return addClass<wfa_map::tlvClientAssociationControlRequest>();
+        }
+        case (160):{
+        return addClass<wfa_map::tlvHigherLayerData>();
+        }
+        case (161):{
+        return addClass<wfa_map::tlvApCapability>();
+        }
+        default:{
+        return addClass<tlvUnknown>();
+        }
+    }
+}
+
+bool CmduMessageRx::parse(uint8_t *buff, size_t buff_len, bool swap_needed, bool parse_tlvs)
```

Comment:
If you return a bool, you need to check if some places on the code still refer to the previous cmdu_header return value.
For example, on beerocks_message.cpp: print_cmdu_type();

Reply:
Since it passed compilation I didn't bother (and since in most cases it is just used to check if the function call was successful - `if (!parse(...))`) - but i'll double check that I didn't miss other places.

Reply:
I looked and its fine, cmdu_header is only used for checking if the function call was successful.